services/src/main/java/org/keycloak/protocol/oidc/endpoints/AuthorizationEndpoint.java

```diff
@@ -384,6 +384,14 @@ private void updateAuthenticationSession() {
                 authenticationSession.setClientNote(LOGIN_SESSION_NOTE_ADDITIONAL_REQ_PARAMS_PREFIX + paramName, request.getAdditionalReqParams().get(paramName));
             }
         }
+
+        // Store authorization_details from authorization/PAR request for later processing
+        if (request.getAdditionalReqParams() != null) {
+            String authorizationDetails = request.getAdditionalReqParams().get(OAuth2Constants.AUTHORIZATION_DETAILS_PARAM);
+            if (authorizationDetails != null) {
+                authenticationSession.setClientNote(OAuth2Constants.AUTHORIZATION_DETAILS_PARAM, authorizationDetails);
```

Comment:
Line 382 and Line 389 do the same thing.
To remove this duplication, how about moving Line 390-392 after Line 385?